src/ui/main.cpp

```diff
@@ -80,20 +231,80 @@ int           main(int, char **) {
     // Load Fonts
     // io.Fonts->AddFontDefault();
 #ifndef IMGUI_DISABLE_FILE_FUNCTIONS
-    io.Fonts->AddFontFromFileTTF("fonts/Roboto-Medium.ttf", 16.0f);
+    // io.Fonts->AddFontFromFileTTF("fonts/Roboto-Medium.ttf", 16.0f);
     // io.Fonts->AddFontFromFileTTF("fonts/Cousine-Regular.ttf", 15.0f);
     // io.Fonts->AddFontFromFileTTF("fonts/DroidSans.ttf", 16.0f);
     // io.Fonts->AddFontFromFileTTF("fonts/ProggyTiny.ttf", 10.0f);
     // ImFont* font = io.Fonts->AddFontFromFileTTF("fonts/ArialUni.ttf", 18.0f, NULL, io.Fonts->GetGlyphRangesJapanese());
     // IM_ASSERT(font != NULL);
 #endif
 
+    App app = {
+        .flowGraph = {},
+        .fgItem    = { &app.flowGraph },
+        .dashboard = DigitizerUi::Dashboard(&app.flowGraph)
+    };
+
+    app.fgItem.newSinkCallback = [&, n = 1]() mutable {
+        auto name = fmt::format("sink {}", n++);
+        app.flowGraph.addSinkBlock(std::make_unique<DigitizerUi::DataSink>(name));
+    };
+
+#ifndef EMSCRIPTEN
+    app.flowGraph.loadBlockDefinitions(BLOCKS_DIR);
+#endif
+    app.flowGraph.parse(opencmw::URI<opencmw::STRICT>("http://localhost:8080/flowgraph"));
```

Comment:
fetching the flowgraph on emscripten is disabled because of the missing http client in opencmw, will be fixed when the emscripten capable client is integrated.